speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionsTests.cpp

```diff
@@ -0,0 +1,121 @@
+
+    SMediator::GetInstance()->CarController = carController;
+}
+
+/// @brief Tests if all decisions do their RunInterveneCommand correctly
+TEST(DecisionsTest, RunInterveneDecisions)
+{
+    InitializeMediator();
+
+    Random random;
+
+    BrakeDecision brakeDecision;
+    float controlBrakeAmount = random.NextFloat();
+    brakeDecision.BrakeAmount = controlBrakeAmount;
+    brakeDecision.RunInterveneCommands();
+
+    std::cout << "Testing brake...";
+    ASSERT_ALMOST_EQ(controlBrakeAmount, SMediator::GetInstance()->CarController.GetBrakeCmd(), 0.001f);
+    std::cout << " check" << std::endl;
+
+    AccelDecision accelDecision;
+    float controlAccelAmount = random.NextFloat();
+    accelDecision.AccelAmount = controlAccelAmount;
+    accelDecision.RunInterveneCommands();
+
+    std::cout << "Testing accel...";
+    ASSERT_ALMOST_EQ(controlAccelAmount, SMediator::GetInstance()->CarController.GetAccelCmd(), 0.001f);
+    std::cout << " check" << std::endl;
+
+    SteerDecision steerDecision;
+    float controlSteerAmount = random.NextFloat();
+    steerDecision.SteerAmount = controlSteerAmount;
+    steerDecision.RunInterveneCommands();
+
+    std::cout << "Testing steer...";
+    ASSERT_ALMOST_EQ(controlSteerAmount, SMediator::GetInstance()->CarController.GetSteerCmd(), 0.001f);
+    std::cout << " check" << std::endl;
+
+    std::cout << "Checking if no value was changed that should not have been changed...";
+    ASSERT_ALMOST_EQ(controlBrakeAmount, SMediator::GetInstance()->CarController.GetBrakeCmd(), 0.001f);
+    ASSERT_ALMOST_EQ(controlAccelAmount, SMediator::GetInstance()->CarController.GetAccelCmd(), 0.001f);
+    ASSERT_ALMOST_EQ(controlSteerAmount, SMediator::GetInstance()->CarController.GetSteerCmd(), 0.001f);
+    std::cout << " check" << std::endl;
+}
+
+/// @brief Checks if the brake decision RunIndicateCommand works correctly
+TEST(DecisionTests, BrakeRunIndicateTest)
+{
+    InitializeMediator();
+    GfInit();
+
+    BrakeDecision brakeDecision;
+    brakeDecision.BrakeAmount = 1;
+    brakeDecision.RunIndicateCommands();
+
+    auto activeIndicators = IndicatorConfig::GetInstance()->GetActiveIndicators();
+
+    ASSERT_EQ(activeIndicators[0].Action, 3);
+}
+
+/// @brief Checks if the steer decision RunIndicateCommand works correctly
+TEST(DecisionsTest, SteerRunIndicateTests)
+{
+    InitializeMediator();
+    GfInit();
+
+    SteerDecision steerDecision;
+    steerDecision.SteerAmount = -1;
+    steerDecision.RunIndicateCommands();
+
+    // TODO: Update to have multiple indicators when indicator code is updated
+    auto activeIndicators = IndicatorConfig::GetInstance()->GetActiveIndicators();
+
+    ASSERT_EQ(activeIndicators[0].Action, INTERVENTION_ACTION_TURN_RIGHT);
+
+    steerDecision.SteerAmount = 1;
+    steerDecision.RunIndicateCommands();
+
+    activeIndicators = IndicatorConfig::GetInstance()->GetActiveIndicators();
+
+    ASSERT_EQ(activeIndicators[0].Action, INTERVENTION_ACTION_TURN_LEFT);
+}
+
```

Comment:
Missing a test for multiple activeindicators or just anything that tests more than just the first element in the activeIndicators array

Reply:
Due to how the code behaves currently, there will never be more than 1 element in the vector